packages/paste-website/src/pages/components/form/index.mdx

```diff
@@ -0,0 +1,690 @@
+- If the label is longer than 3 words/20 characters, consider using a vertically-stacked layout.
+- If the form element’s content is too long, use a single-column, vertical layout.
+
+<LivePreview
+  scope={{
+    Form,
+    FormActions,
+    FormControl,
+    Heading,
+    Paragraph,
+    Radio,
+    RadioGroup,
+    Checkbox,
+    CheckboxGroup,
+    Button,
+    useUIDSeed,
+  }}
+  language="jsx"
+  noInline
+>
+  {horizontalForm}
+</LivePreview>
+
+### Sections
+
+Use sections to group related content or steps within a form for users to input. A section is separated using a [Separator](/components/separator), with `$space-90` above and below it. Sections can also include a section heading and description, if applicable. If a form element doesn’t belong to any of the sections, make sure there’s enough spacing between to signal that the form element is separate doesn’t belong to the section, with at least `$space-130`.
+
+<LivePreview
+  scope={{
+    Form,
+    FormActions,
+    FormControl,
+    FormSection,
+    FormSectionDescription,
+    FormSectionHeading,
+    Separator,
+    SearchIcon,
+    Heading,
+    Input,
+    Label,
+    Button,
+    Select,
+    Option,
+    useUIDSeed,
+    HelpText,
+    Paragraph,
+  }}
+  language="jsx"
+  noInline
+>
+  {sectionsForm}
+</LivePreview>
+
+### Form actions
+
+A form can include form actions if placed within a page or page section. Using form actions helps keep the form elements left aligned for better scannability of the content and its related actions. Don't prevent form submission by disabling the submit button.
+
+<LivePreview scope={{FormActions, Button}} language="jsx" noInline>
+  {formActionsExample}
+</LivePreview>
+
+### Validation
+
+Use validation and [error messaging](/components/form#errors) to indicate when a form submission fails or requires additional information to be shown.
+
+Validate form fields on form submission. Validating a form field when the user leaves the current field (on blur) can be helpful to check for syntax errors. However, this can be frustrating to users who tab through controls to navigate a page, and to screen reader users, who might not be able to detect that an error occurred on blur.
+
+Don't prevent form submission by disabling the submit button. An error message can give more information than a disabled button can to help users figure out which fields are invalid.
+
+#### Errors
+
+Use inline error messages to inform users when they cannot continue completing a form, then provide guidance on next steps and ways to remedy the situation.
+
+To provide further clarity, doubling on the use of [Callouts](/components/callout) and inline error messages helps users quickly notice, find, and remedy errors. If there are multiple errors, summarize them in a logical order (e.g. for forms that are missing several required fields from the user). Callouts should be placed below the Form title or section title, if applicable, and above the form elements.
```

Comment:
I wanted to add in how folks can also use toasts, as this recently came up as a question in a GH discussion and previous GH discussions as well.

New paragraph suggestion: If your form lives within a Modal or Side Modal and has a form action to submit or save the information, we suggest closing the modal on submit, then display a Toast that indicates the action was successful or experienced a system error.